### ISSUE TITLE
[Suggestion] persistent custom config support

### ISSUE THREAD

**Issue by comuslxl** [kingToolbox/WindTerm#89]
I know that most of configs can be modified by .config file, but here comes a problem for version update. After version update, all of the modified configs are overwritten by the new default ones. I cannot just backup the older configs and replace the new ones because there might be some new configurable items in the new config files.

It'll be better to have a standalone folder, eg custom/, to store custom config and in this way I can take a backup-restore operation to keep my custom config. Thank you!

**Comment by kingToolbox** [comment]
Thank you. If you don't modify the keyboard or themes files, the upgrade should be very simple. I think it looks like you might have missed the instructions in the [release](https://github.com/kingToolbox/WindTerm/releases/tag/2.1)? :p

> Upgrade
> 
> Copy or move the "profiles" folder from the old Windows/Linux version to the new Windows/Linux version and all the sessions and configuration will be restored.

**Comment by comuslxl** [comment]
I DO have read the the release note, and what I want to keep is excatly the shortcut config and some other configs in "global/" folder, eg application.lockScreenTimeout and text.autoCopySeletion and something else。😆

**Comment by kingToolbox** [comment]
Sorry, I didn't figure it out. Manual modifications to configuration items, keyboards, etc. are indeed a problem when upgrading. I have determined that the next version 2.2.0 in May will improve the settings, including providing an interface, and modifying part of the design to make it independent and easy to upgrade, as you described. But now you may still need to do it manually. 

First of all, it is not recommended to modify `global/wind.config` directly. A better way is to modify `profiles\default.v10\user.config`. The value of each item in `user.config` will overwrite the value in `wind.config`. Take the `application.lockScreen: 1800` in `wind.config` as an example, you only need to add `application.lockScreen: 600` to `user.config`, then the value will be 600 instead of 1800 at runtime. This is also how WindTerm stores user configurations. All configuration items that provide a modification entry in WindTerm will be automatically output to `user.config` after modification, like `text.zoomFactor` and so on, so that you can easily upgrade by copying the `profile` folder.

For the other configuration files you may have modified, like `wind.keymaps` config file, I am afraid there is currently no such overwrite mechanism, so you can only merge it manually.

**Comment by comuslxl** [comment]
>A better way is to modify profiles\default.v10\user.config. The value of each item in user.config will overwrite the value in wind.config. 

I've tested user.config & wind.config overwrite policy, and it works. 

>For the other configuration files you may have modified, like wind.keymaps config file, I am afraid there is currently no such overwrite mechanism, so you can only merge it manually.

Do you have any plan for this feature?

**Comment by kingToolbox** [comment]
Yes, at least the GUI of all known configuration items in `king.config` and shortcuts will be provided in the version 2.2.0 in May. In addition, an automatic upgrade tool will be provided in the near future, and some configurations will be automatically migrated from the old version to the new version in this tool.

Other configurations, such as themes, will wait until the plug-in system is implemented and use the same upgrade method as the plug-in. This is expected to wait until the second half of this year.

**Comment by comuslxl** [comment]
That will be great! Thank you!

**Comment by kingToolbox** [comment]
Hello, the Settings dialog `Menubar - Session (1) - Preferences - Settings` has been added in 2.2.0 prerelease versions. Please download and check the new version [WindTerm_2.2.0_Prerelease_3](https://github.com/kingToolbox/WindTerm/releases/tag/2.2-prerelease). Except for the modification of shortcut keys, you can now use the interface to conveniently set up the rest of the configuration. 

The shortcut keys setting interface will be provided in subsequent versions. Please stay tuned, Thank you.
